src/hotspot/cpu/x86/sharedRuntime_x86_64.cpp

```diff
@@ -1592,55 +1546,23 @@ nmethod* SharedRuntime::generate_native_wrapper(MacroAssembler* masm,
   // the hidden arguments as arg[0] and possibly arg[1] (static method)
 
   const int total_in_args = method->size_of_parameters();
-  int total_c_args = total_in_args;
-  if (!is_critical_native) {
-    total_c_args += 1;
-    if (method->is_static()) {
-      total_c_args++;
-    }
-  } else {
-    for (int i = 0; i < total_in_args; i++) {
-      if (in_sig_bt[i] == T_ARRAY) {
-        total_c_args++;
-      }
-    }
+  int total_c_args = total_in_args+1;
+  if (method->is_static()) {
+    total_c_args++;
```

Comment:
In this patch, sometimes we keep the if structure, like here, but in other places, we replace this with:
```
  int total_c_args = total_in_args + (method->is_static() ? 2 : 1)
```
Should probably stick with a single style.

Reply:
Ok, that's a good suggestion.  Some platforms have a method_is_static boolean and some don't, so I didn't clean up the platforms that do that later in a different way (or inconsistently).